### PR TITLE
OTA - Add private function to free buffers for unhandled events

### PR DESCRIPTION
**Pull request by pvyawaha**
The change add a private function to free buffers for unexpected job and data events. The change also fixes file close when OTA resumes.

Description
-----------

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [ ] My code is Linted.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: libraries/freertos_plus/aws/ota/src/aws_iot_ota_agent.c [hunk 1]
```diff
@@ -857,11 +861,11 @@ static OTA_Err_t prvProcessJobHandler( OTA_EventData_t * pxEventData )
         else
         {
             /*
-             * If the job context returned NULL and the image state is not in the self_test state,
-             * then an error occurred parsing the OTA job message.  Abort the OTA job with a parse error.
-             *
-             * If there is a valid job id, then a job status update will be sent.
-             */
+              * If the job context returned NULL and the image state is not in the self_test state,
```
Comment (ravibhagavandas) [inline libraries/freertos_plus/aws/ota/src/aws_iot_ota_agent.c:864]: Are these uncrustify related changes ?

File: libraries/freertos_plus/aws/ota/src/aws_iot_ota_agent.c [hunk 2]
```diff
@@ -73,19 +73,19 @@ typedef struct OTAStateTableEntry
...
- * null pointer. No structure is anywhere near 64K in size.
- */
+  * Returns the byte offset of the element 'e' in the typedef structure 't'.
+  * Setting an arbitrarily large base of 0x10000 and masking off that base allows
+  * us to do the same thing as a zero offset without the lint warnings of using a
+  * null pointer. No structure is anywhere near 64K in size.
+  */
 /*lint -emacro((923,9078),OFFSET_OF) Intentionally cast pointer to uint32_t because we are using it as an offset. */
 
 #define OFFSET_OF( t, e )    ( ( uint32_t ) ( &( ( t * ) 0x10000UL )->e ) & 0xffffUL )
```
Comment (gkwicker) [inline libraries/freertos_plus/aws/ota/src/aws_iot_ota_agent.c:83]: Why are we reinventing `offsetof()`?  It is [available in C89.](http://man7.org/linux/man-pages/man3/offsetof.3.html)

File: libraries/freertos_plus/aws/ota/src/aws_iot_ota_agent.c (hunk 1 above)
Comment (gkwicker) [reply libraries/freertos_plus/aws/ota/src/aws_iot_ota_agent.c:864]: Looks like it

File: libraries/freertos_plus/aws/ota/src/aws_iot_ota_agent.c [hunk 3]
```diff
@@ -2399,6 +2404,42 @@ static void prvAgentShutdownCleanup( void )
     }
 }
 
+/*
+  * Handle any events that were unexpected in the current state.
+  */
+static void prvHandleUnexpectedEvents( OTA_EventMsg_t * pxEventMsg )
+{
+    DEFINE_OTA_METHOD_NAME( "prvHandleUnexpectedEvents" );
```
Comment (gkwicker) [inline libraries/freertos_plus/aws/ota/src/aws_iot_ota_agent.c:2412]: Consider `configASSERT( pxEventMsg );`

File: libraries/freertos_plus/aws/ota/src/aws_iot_ota_agent.c [hunk 4]
```diff
@@ -2399,6 +2404,42 @@ static void prvAgentShutdownCleanup( void )
...
+        case eOTA_AgentEvent_ReceivedFileBlock:
+
+            /* Received file data event is not handled , release the buffer.*/
+            prvOTAEventBufferFree( pxEventMsg->pxEventData );
+
+            break;
+
+        default:
+
+            /* Nothing to do here.*/
```
Comment (gkwicker) [inline libraries/freertos_plus/aws/ota/src/aws_iot_ota_agent.c:2438]: Wouldn't you always want to log here?  It's an unhandled unexpected event, and logging could be disabled on line 2414, correct?

Comment (pvyawaha) [reply libraries/freertos_plus/aws/ota/src/aws_iot_ota_agent.c:2438]: On line 2414 we are logging all the unhandled events and what state OTA Agent is, then for some events we do required cleanup. So all the events are anyway unexpected event, moving the logging to default will require us to add logging other cases as well , but with same information.

File: libraries/freertos_plus/aws/ota/src/aws_iot_ota_agent.c (hunk 3 above)
Comment (pvyawaha) [reply libraries/freertos_plus/aws/ota/src/aws_iot_ota_agent.c:2402]: Changed in https://github.com/aws/amazon-freertos/pull/2025/commits/ec0eaa30008d145e3809864d7f6ce6f06f3b6f0a

File: libraries/freertos_plus/aws/ota/src/aws_iot_ota_agent.c (hunk 2 above)
Comment (pvyawaha) [reply libraries/freertos_plus/aws/ota/src/aws_iot_ota_agent.c]: Fixed in https://github.com/aws/amazon-freertos/pull/2025/commits/ec0eaa30008d145e3809864d7f6ce6f06f3b6f0a

File: libraries/freertos_plus/aws/ota/src/aws_iot_ota_agent.c (hunk 1 above)
Comment (pvyawaha) [reply libraries/freertos_plus/aws/ota/src/aws_iot_ota_agent.c:854]: Yes , they are with the latest config.

File: libraries/freertos_plus/aws/ota/src/aws_iot_ota_agent.c (hunk 3 above)
Comment (gkwicker) [reply libraries/freertos_plus/aws/ota/src/aws_iot_ota_agent.c:2402]: nice
